edb/edgeql/compiler/viewgen.py

```diff
@@ -613,6 +613,8 @@ def _normalize_view_ptr_expr(
                 is_update=is_update,
             )
 
+            # Mutation shapes don't obey the same rules as select shapes.
+            shape_expr_ctx.in_shape = not is_mutation
```

Comment:
This seems superfluous.  Use `defining_view` below instead: `ctx.defining_view is not None and ctx.defining_view.get_expr_type() is s_types.ExprType.Select`